libs/zephyr-agent/src/lib/errors/global-catcher.ts

```diff
@@ -0,0 +1,23 @@
+import { logFn } from '../logging/ze-log-event';
+import { ZephyrError } from './zephyr-error';
+
+/**
+ * Wraps async function with error handling, logs errors automatically. Set
+ * ZE_FAIL_BUILD=true to throw errors instead of logging and returning fallback.
+ */
+export async function catchAsync<T>(
+  fn: () => Promise<T>,
+  fallback?: T
+): Promise<T | undefined> {
+  try {
+    return await fn();
+  } catch (error) {
+    if (process.env['ZE_FAIL_BUILD'] === 'true') {
+      throw error;
+    }
+
+    logFn('error', ZephyrError.format(error));
+
+    return fallback;
+  }
+}
```

Comment:
This is a very textbok example to use an Either<T, E>

`type Result<T> = { _tag: "error" ; value: T } | { _tag: "success"; error: Error };`

Reply:
Another note that comes to mind is to use overloads for these, so they're actually typesafe. For the return types.

Reply:
Added the overloads